agenda/domain/src/main/java/com/pronoidsoftware/agenda/domain/model/AgendaItemType.kt

```diff
@@ -0,0 +1,24 @@
+package com.pronoidsoftware.agenda.domain.model
+
+enum class AgendaItemType(private val type: String) {
+
+    EVENT("EVENT"),
+    TASK("TASK"),
+    REMINDER("REMINDER"),
+    ;
+
+    override fun toString(): String {
+        return type
+    }
+
+    companion object {
+        fun from(type: String): AgendaItemType {
+            return when (type) {
+                "EVENT" -> EVENT
+                "TASK" -> TASK
+                "REMINDER" -> REMINDER
+                else -> error("Unknown Agenda Item type")
+            }
+        }
+    }
+}
```

Comment:
What you implemented enums already have built in:
```
val enum = AgendaItemType.EVENT
val enumString = enum.name
val enumAgain = AgendaItemType.valueOf(enumString) 
```

Reply:
ah, thanks for pointing this out 👍